recipes/jerryscript/all/conanfile.py

```diff
@@ -41,7 +115,36 @@ def configure(self):
         del self.settings.compiler.cppstd
         if self.options.shared:
             del self.options.fPIC
-            
+
+        if not self.options.debugger:
+            del self.options.keep_line_info
+        
+        if self.options.profile is None:
+            if Version(self.version) < Version("2.4.0"):
+                self.options.profile = "es5.1"
+            else:
+                self.options.profile = "es.next"
+        if (not self.options.profile in self.predefined_profiles) and not os.path.isfile(self.options.profile):
+            raise ConanInvalidConfiguration("jerryscript feature profile must either be a valid file or one of these: es.next, es5.1, minimal")
+        
+        try:
+            assert(int(self.options.heap_size) >= 0), "must be bigger than or equal to 0"
+            assert(int(self.options.gc_mark_limit) >= 0), "must be bigger than or equal to 0"
+            assert(int(self.options.stack_limit) >= 0), "must be bigger than or equal to 0"
+        except:
+            raise ConanInvalidConfiguration("jerryscript heap size, gc mark limit, stack limit should be a positive integer")
+
+        if self.options.jerry_math is None:
+            if Version(self.version) < Version("2.4.0"):
+                self.options.jerry_math = True
+            else:
+                self.options.jerry_math = False
+
+        if self.settings.arch == "x86_64":
+            self.options.system_allocator = False
+        if self.options.system_allocator:
+            self.options.cpointer_32_bit = True
```

Comment:
This overrides a user option.
Is `cpointer_32_bit` actually useful?
I mean, isn't it always 32-bit on x86 and 64-bit on x86_64?
Same for armv7 vs armv8?

Reply:
This library operates a javascript engine, and you can manage the pointer size for the bytecode it generates when targeting devices with smaller footprints. It will of course work as you say when the system allocator is used. Should I instead throw an exception here if the validation fails?

Reply:
Just never override user options. It's very surprising for a user if the library is not compiled as expected.
Try to set a good default in `config_options` and do some basic tests in `validate` whether an option is ok.

It's ok to let a user pass invalid options (arch=x86_64 system_allocator=True and cpointer_32_bit=True), and let validate throw an error.